src/action/AsyncUserActions.ts

```diff
@@ -129,6 +130,44 @@ export function register(user: UserAccountData) {
   };
 }
 
+/**
+ * Creates a request to change a password of account with provided username.
+ * @param username The username specifying user account
+ */
+export function requestPasswordReset(username: string) {
+  const action = {
+    type: ActionType.REQUEST_PASSWORD_RESET,
+  };
+  return (dispatch: ThunkDispatch) => {
+    dispatch(asyncActionRequest(action));
+    return Ajax.post(
+      Constants.API_PREFIX + "/password",
+      content(username).contentType("text/plain")
+    )
+      .then(() => dispatch(asyncActionSuccess(action)))
+      .catch((error: ErrorData) => dispatch(asyncActionFailure(action, error)));
+  };
+}
+
+/**
+ * uses password reset token to change password of associated account
+ * @param dto ChangePasswordDto with required data
+ */
+export function resetPassword(dto: ChangePasswordDto) {
+  const action = {
+    type: ActionType.RESET_PASSWORD,
+  };
+  return (dispatch: ThunkDispatch) => {
+    dispatch(asyncActionRequest(action));
+    return Ajax.put(
+      Constants.API_PREFIX + "/password",
+      content(dto).contentType("application/json")
```

Comment:
This should be JSON-LD and the data must contain the corresponding _context_

I also forgot that the backend DTO must thus also be annotated with mapping annotations so that JSON-LD can be mapped to them as well.